kotlin-react/src/main/kotlin/react/hooks.kt

```diff
@@ -8,48 +8,38 @@ typealias RDependenciesList = List<dynamic>
 
 typealias RSetState<T> = (value: T) -> Unit
 
-fun <T> useState(initValue: T): Pair<T, RSetState<T>> {
+fun <T> useState(initValue: T): RStateDelegate<T> {
     val jsTuple = rawUseState(initValue)
-    val currentValue = jsTuple[0] as T
-    val setState = jsTuple[1] as RSetState<T>
-    return currentValue to setState
+    return RStateDelegate(
+        jsTuple[0] as T,
+        jsTuple[1] as RSetState<T>
+    )
 }
 
-fun <T> useState(valueInitializer: () -> T): Pair<T, RSetState<T>> {
+fun <T> useState(valueInitializer: () -> T): RStateDelegate<T> {
     val jsTuple = rawUseState(valueInitializer)
-    val currentValue = jsTuple[0] as T
-    val setState = jsTuple[1] as RSetState<T>
-    return currentValue to setState
+    return RStateDelegate(
+        jsTuple[0] as T,
+        jsTuple[1] as RSetState<T>
+    )
 }
 
-operator fun <T> Pair<T, RSetState<T>>.getValue(thisRef: Any?, property: KProperty<*>): T {
-    return first
-}
-
-operator fun <T> Pair<T, RSetState<T>>.setValue(thisRef: Any?, property: KProperty<*>, value: T) {
-    second(value)
-}
+class RStateDelegate<T>
+internal constructor(
+    private val state: T,
+    private val setState: RSetState<T>
+) : ReadWriteProperty<Any, T> {
+    operator fun component1(): T = state
+    operator fun component2(): RSetState<T> = setState
 
-private class ReactStateDelegate<T>(useState: Pair<T, RSetState<T>>) : ReadWriteProperty<Any?, T> {
-    private val state = useState.first
-    private val setState = useState.second
-
-    override operator fun getValue(thisRef: Any?, property: KProperty<*>): T =
+    override operator fun getValue(thisRef: Any, property: KProperty<*>): T =
```

Comment:
Are you sure that changing `Any?` on `Any` is valid?
Seems that it leads to error, because this delegate can be used in function scope without `thisRef`

Reply:
Fixed